library/common/main_interface.cc

```diff
@@ -121,23 +151,28 @@ envoy_status_t register_platform_api(const char* name, void* api) {
   return ENVOY_SUCCESS;
 }
 
-/**
- * External entrypoint for library.
- */
-envoy_status_t run_engine(envoy_engine_t, envoy_engine_callbacks callbacks, envoy_logger logger,
-                          const char* config, const char* log_level) {
+envoy_engine_t init_engine(envoy_engine_callbacks callbacks, envoy_logger logger) {
+  // TODO(goaway): return new handle once multiple engine support is in place.
+  // https://github.com/lyft/envoy-mobile/issues/332
+  strong_engine_ = std::make_shared<Envoy::Engine>(callbacks, logger, preferred_network_);
+  engine_ = strong_engine_;
+  return 1;
+}
+
+envoy_status_t run_engine(envoy_engine_t, const char* config, const char* log_level) {
   // This will change once multiple engine support is in place.
   // https://github.com/lyft/envoy-mobile/issues/332
 
-  // The shared pointer created here will keep the engine alive until static destruction occurs.
-  strong_engine_ =
-      std::make_shared<Envoy::Engine>(callbacks, logger, config, log_level, preferred_network_);
+  if (auto e = engine()) {
+    e->run(config, log_level);
+    return ENVOY_SUCCESS;
+  }
 
-  // The weak pointer we actually expose allows calling threads to atomically check if the engine
-  // still exists and acquire a shared pointer to it - ensuring the engine persists at least for
-  // the duration of the call.
-  engine_ = strong_engine_;
-  return ENVOY_SUCCESS;
+  return ENVOY_FAILURE;
 }
 
-void terminate_engine(envoy_engine_t) { strong_engine_.reset(); }
+void terminate_engine(envoy_engine_t) {
+  auto e = strong_engine_;
```

Comment:
leave comment explaining the local var? Or maybe it is self-explanatory?

Reply:
Sure.